flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/TestingJobManagerRunnerRegistry.java

```diff
@@ -188,11 +197,26 @@ public static class Builder {
         private Supplier<Collection<JobManagerRunner>> getJobManagerRunnersSupplier =
                 Collections::emptyList;
         private Function<JobID, JobManagerRunner> unregisterFunction = ignoredJobId -> null;
-        private BiFunction<JobID, Executor, CompletableFuture<Void>> localCleanupAsyncFunction =
-                (ignoredJobId, ignoredExecutor) -> FutureUtils.completedVoidFuture();
+        private TriFunction<JobID, Executor, Executor, CompletableFuture<Void>>
+                localCleanupAsyncFunction =
+                        (ignoredJobId, ignoredExecutor, mainThreadExecutor) ->
+                                FutureUtils.completedVoidFuture();
         private BiFunction<JobID, Executor, CompletableFuture<Void>> globalCleanupAsyncFunction =
                 (ignoredJobId, ignoredExecutor) -> FutureUtils.completedVoidFuture();
 
+        private Builder fromDefaultJobManagerRunnerRegistry(
```

Comment:
What's the purpose of this? It looks like we're map each of the `DefaultJobManagerRunnerRegistry` methods to the `TestingJobManagerRunnerRegistry` callback generating actually a "new" `DefaultJobManagerRunnerRegistry` instance wrapped in a `TestingJobManagerRunnerRegistry`. We could just use `DefaultJobManagerRunnerRegistry`, instead. Don't you think? :thinking: